genie-agent/src/main/java/com/netflix/genie/agent/execution/services/AgentJobService.java

```diff
@@ -51,6 +53,17 @@ void handshake(
         @Valid AgentClientMetadata agentClientMetadata
     ) throws HandshakeException;
 
+    /**
+     * Obtain server-provided configuration properties.
+     *
+     * @param agentClientMetadata metadata about the client making this request
+     * @throws ConfigureException if the server properties cannot be obtained
+     * @return a map of properties
+     */
+    Map<String, String> configure(
+        @Valid AgentClientMetadata agentClientMetadata
```

Comment:
Is this sending the same payload as the `handshake`? I assume it's separate as the handshake can happen multiple times with different servers but this should only happen once per job?

Reply:
Yes it's the same payload.
Basically allowing the server to send down different properties based on agent version, if that ever becomes necessary.

(Handshake is also executed once per job, not once per connection)